subiquity/server/controllers/install.py

```diff
@@ -130,12 +160,97 @@ async def configure_apt(self, *, context):
         configurer = await mirror.wait_config()
         return await configurer.configure_for_install(context)
 
+    async def run_curtin_install_step(
+            self, step: CurtinInstallStep,
+            resume_data_file: Path,
+            context, source) -> subprocess.CompletedProcess:
+        """ Run a curtin install step. """
+
+        self.app.note_file_for_apport(
+                f"Curtin{step.name}Config", str(step.config_file))
+        self.app.note_file_for_apport(
+                f"Curtin{step.name}Errors", str(step.error_file))
+        self.app.note_file_for_apport(
+                f"Curtin{step.name}Log", str(step.log_file))
+
+        self.write_config(
+                config_file=step.config_file,
+                config=step.acquire_config(step, resume_data_file)
+                )
+
+        # Make sure the log directory exists.
+        step.log_file.parent.mkdir(parents=True, exist_ok=True)
+
+        return await run_curtin_command(
+                self.app, context,
+                "install", source,
+                "--set", f'json:stages={json.dumps(step.stages)}',
+                config=str(step.config_file),
+                private_mounts=False)
+
     @with_context(
         description="installing system", level="INFO", childlevel="DEBUG")
     async def curtin_install(self, *, context, source):
-        await run_curtin_command(self.app, context, 'install', source,
-                                 config=self.write_config(),
-                                 private_mounts=False)
+        # Until https://bugs.launchpad.net/curtin/+bug/1876984 gets
+        # fixed, the only way to get curtin to leave the network
+        # config entirely alone is to omit the 'network' stage.
```

Comment:
I don't think this comment really belongs any more? We are moving to a model where we include things in stages rather than exclude them, I guess.

Reply:
Right, I'll drop this comment. Thanks.